lib/components/providers/theme/themeContext.tsx

```diff
@@ -0,0 +1,8 @@
+import React, { type Dispatch, type SetStateAction } from "react";
+
+export const initialThemeState = {
+    theme: "light",
+    setTheme: (() => null) as Dispatch<SetStateAction<string>>,
```

Comment:
```suggestion
    type Theme = 'dark' | 'light';
    setTheme: (() => null) as Dispatch<SetStateAction<Theme>>,
```

Make it more Typesafe